test/tests/functional/pbs_cgroups_hook.py

```diff
@@ -2618,17 +2638,76 @@ def test_cgroup_msmom_nodes_only(self):
         jid2 = self.server.submit(j2)
         self.server.expect(JOB, {ATTR_state: 'R'}, id=jid2)
 
+    def test_cgroups_abort(self):
+        """
+        Verify that if one of the sister mom is down then
+        cgroups hook will call the abort event which will
+        cleanup the cgroups files on sister moms and primary
+        mom
+        """
+
+        if (len(self.moms) < 3):
+            self.skipTest("Test needs atleast 3 moms")
+
+        self.logger.info("Stopping mom on host %s" % self.host[1])
+        self.mom_l[1].signal('-19')
+
+        a = {'Resource_List.select':
+             '1:ncpus=1:host=%s+1:ncpus=1:host=%s+1:ncpus=1:host=%s' %
+             (self.host[0], self.host[1], self.host[2])}
+        j = Job(TEST_USER, attrs=a)
+        j.create_script(self.sleep15_job)
+        jid = self.server.submit(j)
+        a = {'job_state': 'R', 'substate': '41'}
+        self.server.expect(JOB, a, jid)
+
+        self.logger.info("Killing mom on host %s" % self.host[1])
+        now = int(time.time())
+        self.mom_l[1].signal('-9')
+
+        self.server.expect(NODE, {'state': "down"}, id=self.host[1])
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid)
+
+        # Verify that cgroups directories are cleaned on primary mom
+        cpath = self.get_cgroup_job_dir('memory', jid, self.host[0])
+        self.assertFalse(self.is_dir(cpath, self.host[0]))
+
+        # Verify that cgroups directories are cleaned by execjob_abort
+        # hook on sister mom
+        cpath = self.get_cgroup_job_dir('memory', jid, self.host[2])
+        self.assertFalse(self.is_dir(cpath, self.host[2]))
+
+        self.mom_l[0].log_match("job_start_error", starttime=now)
+        self.mom_l[0].log_match("Event type is execjob_abort", starttime=now)
+        self.mom_l[0].log_match("Event type is execjob_epilogue",
+                                starttime=now)
+        self.mom_l[0].log_match("Event type is execjob_end", starttime=now)
+        self.mom_l[2].log_match("Event type is execjob_abort", starttime=now)
+
+        self.mom_l[1].restart()
```

Comment:
@mike0042 since we have stopped mom[1] it won't have anything from that job. we needed 3 moms to have job_start_error followed by exec job_abort on sister mom to verify the bug. in buggy case there will not be any cleanup done by exec job_periodic hook on sister moms leaving the cgroups job directories around.

Reply:
Thanks for clarifying.